### ISSUE TITLE
Slightly odd behaviour of make_first_level_design_matrix

### ISSUE THREAD

**Issue by rhodricusack** [nilearn/nilearn#2668]
nilearn 0.70

I discovered a slightly unexpected behaviour of nilearn.glm.first_level.make_first_level_design_matrix which is that if an event is duplicated with the same onset and duration, one is ignored, but if they differ (even very slightly) then both count towards the design. I expected it to add together multiple events. Code to reproduce below - in X0, events with onsets at 0 and 10 s have same amplitude, whereas in X0, first has twice the amplitude.

Thanks so much for all of your work on this package - it is awesome.

```

from nilearn.glm.first_level import make_first_level_design_matrix
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt


data = [[0, 1, 'cond1'], [0, 1, 'cond1'], [10, 1, 'cond1']] # duplicate event at 0 secs not counted
events = pd.DataFrame(data, columns=['onset','duration','trial_type'])
X0 = make_first_level_design_matrix(np.arange(20),events=events)

data = [[0, 1, 'cond1'], [0.001, 1, 'cond1'], [10, 1, 'cond1']] # duplicate event ~ 0 secs is now counted
events = pd.DataFrame(data, columns=['onset','duration','trial_type'])
X1 = make_first_level_design_matrix(np.arange(20),events=events)

fig, ax = plt.subplots(nrows=2)
ax[0].plot(X0)
ax[1].plot(X1)
plt.savefig('show_make_first_level_design_matrix_oddity.jpg')
``` 
-->

**Comment by bthirion** [comment]
Thx for raising the point. I'm not 100% sure what is the right behavior in that case. I tend to think  that we should indeed leave the use the responsibility to provide an event file that does not contain ambiguous and duplicate entries, and thus not silently ignore some of the provided information: this means that in your case, we should basically add the two events.

Any other opinion ? @tsalo

**Comment by NicolasGensollen** [comment]
I think that adding the events makes the most sense. I would also add a user warning, for example here:

https://github.com/nilearn/nilearn/blob/55e3f26dbd9fc6e89516e5f37e8aae23ec6086a9/nilearn/glm/first_level/experimental_paradigm.py#L21-L24

**Comment by tsalo** [comment]
I agree that overlapping, but distinct, events should be added. @bthirion are you saying that the duplicate events should be treated as distinct ones? Personally, I would lean toward logging a warning when duplicate events are detected, but I think that this is rare enough that either approach should be fine.

**Comment by rhodricusack** [comment]
Great thanks!  To explain two (edge) use cases where we encountered this.
(1) I was testing an efficiency calculation and wanted to see that the calculation on a contrast of two columns was the same as the calculation on a single column made from the sum of the events in the original columns (which it should be if everything is linear). But there were duplicate events in the two columns.

(2) For a naturalistic stimulus, we had two different raters rate what was going on (e.g., is it outside or inside). We then put the events of the two raters into a design, expecting it would give us an "average rating" boxcar values of 0, 1 or 2. But, in the cases where two raters exactly agreed in onset & duration, we lost the duplicates and ended up with 1 not 2.

In both, the easiest behaviour would be to make events add even when they're duplicates.

**Comment by tsalo** [comment]
Thank you, those are both really helpful in understanding the issue. It seems like the question is whether duplicate events should be treated as a valid substitute for amplitude modulation. I can see arguments both against and in favor of this, and honestly I'd be happy going either way, as long as there's a warning logged in either case.

I _am_ curious to know how other packages handle this. I used to use SPM fairly frequently but never had this particular case show up. Does anyone know?

**Comment by bthirion** [comment]
> In both, the easiest behaviour would be to make events add even when they're duplicates.

I think I agree with that, possibly with a warning in case of exact duplicates.

**Comment by NicolasGensollen** [comment]
Addressed in #2680
